entrypoint.sh

```diff
@@ -1,236 +1,227 @@
+    echo -e "WARNING: User with differing UID ${USER_ID}/GID ${GRP_ID} already exists, most likely this container was started before with a different UID/GID. Re-create it to change UID/GID.\n"
+  fi
 else
-    LOCAL_UID=$USER_ID
-    LOCAL_GID=$GRP_ID
-    echo -e "WARNING: Starting container processes as root. This has some security implications and goes against docker best practice.\n"
+  CURRENT_UID="${USER_ID}"
+  CURRENT_GID="${GRP_ID}"
+  echo -e "WARNING: Starting container processes as root. This has some security implications and goes against docker best practice.\n"
 fi
 
-echo -e "Starting with UID/GID : $LOCAL_UID/$LOCAL_GID\n"
+echo -e "Starting with UID/GID : ${CURRENT_UID}/${CURRENT_GID}\n"
 
 # set $HOME
-if [ ! "$USER_ID" == "0"  ]; then
-    export USERNAME=user
-    export HOME=/home/$USERNAME
-else
-    export USERNAME=root
-    export HOME=/root
-fi
-
-# If volumes for zen or zcash-params are present, symlink them to user's home, if not create folders.
-if [ -d "/mnt/zen" ]; then
-    if [ ! -L $HOME/.zen ]; then
-        ln -fs /mnt/zen $HOME/.zen > /dev/null 2>&1
-    fi
+if [ "${CURRENT_UID}" -ne 0 ]; then
+  export USERNAME="user"
+  export HOME="/home/${USERNAME}"
 else
-    mkdir -p $HOME/.zen
-fi
-if [ -d "/mnt/zcash-params" ]; then
-    if [ ! -L $HOME/.zcash-params ]; then
-        ln -fs /mnt/zcash-params $HOME/.zcash-params > /dev/null 2>&1
+  export USERNAME="root"
+  export HOME="/root"
+fi
+
+# check volume mounts are present, symlink to home dir
+mountpoints=( "/mnt/zen" "/mnt/zcash-params" )
+targets=( "${HOME}/.zen" "${HOME}/.zcash-params" )
+for i in "${!mountpoints[@]}"; do
+  while ! mountpoint "${mountpoints[i]}" &> /dev/null; do
+    echo "Waiting for volume ${mountpoints[i]} to be mounted..."
+    sleep 0.5
+  done
+  # ensure there isn't a directory present with the same name or symlink would be created inside of it
+  if ! [ -L "${targets[i]}" ]; then
+    if [ -d "${targets[i]}" ]; then
+        rm -rf "${targets[i]}"
     fi
-else
-    mkdir -p $HOME/.zcash-params
-fi
-
-# Check if we have minimal mainnet and testnet zen.conf files, if not create them.
-if [ ! -e "$HOME/.zen/zen.conf" ]; then
-    touch $HOME/.zen/zen.conf
-fi
-if [ ! -d "$HOME/.zen/testnet3" ]; then
-    mkdir -p $HOME/.zen/testnet3
-fi
-if [ ! -e "$HOME/.zen/testnet3/zen.conf" ]; then
-    touch $HOME/.zen/testnet3/zen.conf
-fi
-if ! grep -q 'rpcuser' $HOME/.zen/zen.conf ; then
-    echo 'rpcuser=user' >> $HOME/.zen/zen.conf
-fi
-if ! grep -q 'rpcuser' $HOME/.zen/testnet3/zen.conf ; then
-    echo 'rpcuser=user' >> $HOME/.zen/testnet3/zen.conf
-fi
-if ! grep -q 'rpcpassword' $HOME/.zen/zen.conf ; then
-    echo "rpcpassword=`head -c 32 /dev/urandom | base64`" >> $HOME/.zen/zen.conf
-fi
-if ! grep -q 'rpcpassword' $HOME/.zen/testnet3/zen.conf ; then
-    echo "rpcpassword=`head -c 32 /dev/urandom | base64`" >> $HOME/.zen/testnet3/zen.conf
-fi
+  fi
```

Comment:
These two if statements could be merged into a single one

Reply:
Done